tensorpack/models/conv2d.py

```diff
@@ -167,27 +208,79 @@ def Conv2DTranspose(
         else:
             kernel_initializer = tf.keras.initializers.VarianceScaling(2.0, distribution='untruncated_normal', seed=seed)
 
-    with rename_get_variable({'kernel': 'W', 'bias': 'b'}):
-        layer = tf.layers.Conv2DTranspose(
-            filters,
-            kernel_size,
-            strides=strides,
-            padding=padding,
-            data_format=data_format,
-            activation=activation,
-            use_bias=use_bias,
-            kernel_initializer=kernel_initializer,
-            bias_initializer=bias_initializer,
-            kernel_regularizer=kernel_regularizer,
-            bias_regularizer=bias_regularizer,
-            activity_regularizer=activity_regularizer,
-            _reuse=tf.get_variable_scope().reuse)
-        ret = layer.apply(inputs, scope=tf.get_variable_scope())
-        ret = tf.identity(ret, name='output')
+    if get_tf_version_tuple() <= (1, 12):
+        with rename_get_variable({'kernel': 'W', 'bias': 'b'}):
+            layer = tf.layers.Conv2DTranspose(
+                filters,
+                kernel_size,
+                strides=strides,
+                padding=padding,
+                data_format=data_format,
+                activation=activation,
+                use_bias=use_bias,
+                kernel_initializer=kernel_initializer,
+                bias_initializer=bias_initializer,
+                kernel_regularizer=kernel_regularizer,
+                bias_regularizer=bias_regularizer,
+                activity_regularizer=activity_regularizer,
+                _reuse=tf.get_variable_scope().reuse)
+            ret = layer.apply(inputs, scope=tf.get_variable_scope())
+            ret = tf.identity(ret, name='output')
+        ret.variables = VariableHolder(W=layer.kernel)
+        if use_bias:
+            ret.variables.b = layer.bias
+    else:
+        # Our own implementation, to avoid Keras bugs. https://github.com/tensorflow/tensorflow/issues/25946
```

Comment:
This comment says 'our own'. I assume it means tensorpack? Maybe you can change the wording

Reply:
Done